osbuild/util/containers.py

```diff
@@ -105,14 +110,30 @@ def merge_manifest(list_manifest, destination):
 
 
 @contextmanager
-def container_source(image):
-    image_filepath = image["filepath"]
-    container_format = image["data"]["format"]
-    image_name = image["data"]["name"]
+def containers_storage_source(image, image_name, image_filepath, container_format):
+    storage_conf = image["data"]["storage"]
+    driver = storage_conf.get("driver", "overlay")
 
-    if container_format not in ("dir", "oci-archive"):
-        raise RuntimeError(f"Unknown container format {container_format}")
+    # use `/containers/storage` for the containers-storage bind mount
+    storage_path = os.path.join(os.sep, "containers", "storage")
+    os.makedirs(storage_path, exist_ok=True)
 
+    with MountGuard() as mg:
+        # NOTE: the ostree.deploy.container needs explicit `rw` access to
+        # the containers-storage store even when bind mounted.
+        mg.mount(image_filepath, storage_path, rw=True)
+
+        image_source = f"{container_format}:[{driver}@{storage_path}+/run/containers/storage]{image_name}"
+        yield image_source
+
+        if driver == "overlay":
+            # NOTE: the overlay sub-directory isn't always released,
+            # so we need to force unmount it
+            subprocess.run(["umount", "-f", os.path.join(storage_path, "overlay")], check=False)
```

Comment:
As an extra measure of protection we could maybe add `--lazy` here so that even if it's busy it will eventually get released